src/main/java/org/pcollections/HashPMap.java

```diff
@@ -134,28 +139,29 @@ public V get(final Object key) {
 	
 	
 //// PRIVATE UTILITIES ////
-	private PSequence<Entry<K,V>> getEntries(final int hash) {
-		PSequence<Entry<K,V>> entries = intMap.get(hash);
+	private ConsPStack<Entry<K,V>> getEntries(final int hash) {
+		ConsPStack<Entry<K,V>> entries = intMap.get(hash);
 		if(entries==null) return ConsPStack.empty();
 		return entries;
 	}
 
 	
 //// PRIVATE STATIC UTILITIES ////
-	private static <K,V> int keyIndexIn(final PSequence<Entry<K,V>> entries, final Object key) {
+	private static <K,V> int keyIndexIn(final ConsPStack<Entry<K,V>> entries, final Object key) {
 		int i=0;
-		for(Entry<K,V> entry : entries) {
+		while(entries != null && !entries.isEmpty()) {
+			Entry<K,V> entry = entries.first;
```

Comment:
This seems to be an infinite loop. `entries` is immutable and final, so the condition of loop never changes.